rules/framework/vfs_overlay.bzl

```diff
@@ -63,6 +63,22 @@ def _make_root(vfs_parent, bin_dir_path, build_file_path, framework_name, root_d
             "external-contents": _get_external_contents(vfs_prefix, module_map[0].path),
         })
 
+    if len(swiftmodule_contents):
+        # Assuming each swift module content share the same parent
+        parent_dir_base_name = swiftmodule_contents[0].dirname.split("/").pop()
+        modules_contents.append({
```

Comment:
what this produces:
```
{
                            "type": "directory",
                            "name": "OTPublishersHeadlessSDK.swiftmodule",
                            "contents": [
                                {
                                    "type": "file",
                                    "name": "arm64-apple-ios-simulator.swiftdoc",
                                    "external-contents": "../../../../../../Pods/OneTrust-CMP-XCFramework/OTPublishersHeadlessSDK.xcframework/ios-arm64_x86_64-simulator/OTPublishersHeadlessSDK.framework/Modules/OTPublishersHeadlessSDK.swiftmodule/arm64-apple-ios-simulator.swiftdoc"
                                },
                                {
                                    "type": "file",
                                    "name": "arm64-apple-ios-simulator.swiftinterface",
                                    "external-contents": "../../../../../../Pods/OneTrust-CMP-XCFramework/OTPublishersHeadlessSDK.xcframework/ios-arm64_x86_64-simulator/OTPublishersHeadlessSDK.framework/Modules/OTPublishersHeadlessSDK.swiftmodule/arm64-apple-ios-simulator.swiftinterface"
                                },
                                {
                                    "type": "file",
                                    "name": "arm64-apple-ios-simulator.swiftinterface-e",
                                    "external-contents": "../../../../../../Pods/OneTrust-CMP-XCFramework/OTPublishersHeadlessSDK.xcframework/ios-arm64_x86_64-simulator/OTPublishersHeadlessSDK.framework/Modules/OTPublishersHeadlessSDK.swiftmodule/arm64-apple-ios-simulator.swiftinterface-e"
                                },
                                {
                                    "type": "file",
                                    "name": "arm64.swiftdoc",
                                    "external-contents": "../../../../../../Pods/OneTrust-CMP-XCFramework/OTPublishersHeadlessSDK.xcframework/ios-arm64_x86_64-simulator/OTPublishersHeadlessSDK.framework/Modules/OTPublishersHeadlessSDK.swiftmodule/arm64.swiftdoc"
                                },
                                {
                                    "type": "file",
                                    "name": "arm64.swiftinterface",
                                    "external-contents": "../../../../../../Pods/OneTrust-CMP-XCFramework/OTPublishersHeadlessSDK.xcframework/ios-arm64_x86_64-simulator/OTPublishersHeadlessSDK.framework/Modules/OTPublishersHeadlessSDK.swiftmodule/arm64.swiftinterface"
                                },
                                {
                                    "type": "file",
                                    "name": "arm64.swiftinterface-e",
                                    "external-contents": "../../../../../../Pods/OneTrust-CMP-XCFramework/OTPublishersHeadlessSDK.xcframework/ios-arm64_x86_64-simulator/OTPublishersHeadlessSDK.framework/Modules/OTPublishersHeadlessSDK.swiftmodule/arm64.swiftinterface-e"
                                },
                                {
                                    "type": "file",
                                    "name": "x86_64-apple-ios-simulator.swiftdoc",
                                    "external-contents": "../../../../../../Pods/OneTrust-CMP-XCFramework/OTPublishersHeadlessSDK.xcframework/ios-arm64_x86_64-simulator/OTPublishersHeadlessSDK.framework/Modules/OTPublishersHeadlessSDK.swiftmodule/x86_64-apple-ios-simulator.swiftdoc"
                                },
                                {
                                    "type": "file",
                                    "name": "x86_64-apple-ios-simulator.swiftinterface",
                                    "external-contents": "../../../../../../Pods/OneTrust-CMP-XCFramework/OTPublishersHeadlessSDK.xcframework/ios-arm64_x86_64-simulator/OTPublishersHeadlessSDK.framework/Modules/OTPublishersHeadlessSDK.swiftmodule/x86_64-apple-ios-simulator.swiftinterface"
                                },
                                {
                                    "type": "file",
                                    "name": "x86_64-apple-ios-simulator.swiftinterface-e",
                                    "external-contents": "../../../../../../Pods/OneTrust-CMP-XCFramework/OTPublishersHeadlessSDK.xcframework/ios-arm64_x86_64-simulator/OTPublishersHeadlessSDK.framework/Modules/OTPublishersHeadlessSDK.swiftmodule/x86_64-apple-ios-simulator.swiftinterface-e"
                                },
                                {
                                    "type": "file",
                                    "name": "x86_64.swiftdoc",
                                    "external-contents": "../../../../../../Pods/OneTrust-CMP-XCFramework/OTPublishersHeadlessSDK.xcframework/ios-arm64_x86_64-simulator/OTPublishersHeadlessSDK.framework/Modules/OTPublishersHeadlessSDK.swiftmodule/x86_64.swiftdoc"
                                },
                                {
                                    "type": "file",
                                    "name": "x86_64.swiftinterface",
                                    "external-contents": "../../../../../../Pods/OneTrust-CMP-XCFramework/OTPublishersHeadlessSDK.xcframework/ios-arm64_x86_64-simulator/OTPublishersHeadlessSDK.framework/Modules/OTPublishersHeadlessSDK.swiftmodule/x86_64.swiftinterface"
                                },
                                {
                                    "type": "file",
                                    "name": "x86_64.swiftinterface-e",
                                    "external-contents": "../../../../../../Pods/OneTrust-CMP-XCFramework/OTPublishersHeadlessSDK.xcframework/ios-arm64_x86_64-simulator/OTPublishersHeadlessSDK.framework/Modules/OTPublishersHeadlessSDK.swiftmodule/x86_64.swiftinterface-e"
                                }
                            ]
                        }
```

Reply:
So two slices are together (arm64 and x86)... which i think is fine because when we do the actual build, only the right interface files will be selected based on the name? Doing tests on downstream project does not seem to have any problem

Reply:
okay i might be wrong... this does not feel right... let me look this a bit, probably some small mistake on the slicing side